src/Entity/Donator.php

```diff
@@ -13,6 +13,7 @@
  *     name="donators",
  *     uniqueConstraints={
  *         @ORM\UniqueConstraint(name="donator_identifier_unique", columns="identifier"),
+ *     *   @ORM\UniqueConstraint(name="donator_unique_matching", columns={"last_name", "first_name", "email_address"}),
```

Comment:
you should respect the order that it'll be used: 
for ex, you have these conditions: 
```
->andWhere('donator.emailAddress = :emailAddress')
->andWhere('donator.firstName = :firstName')
->andWhere('donator.lastName = :lastName')
```

your index should be in this order: `emailAddress, firstName, lastName`